consensus/progpow/poem.go

```diff
@@ -0,0 +1,129 @@
+package progpow
+
+import (
+	"math/big"
+
+	"github.com/dominant-strategies/go-quai/common"
+	"github.com/dominant-strategies/go-quai/core/types"
+	"github.com/dominant-strategies/go-quai/params"
+	"modernc.org/mathutil"
+)
+
+// CalcOrder returns the order of the block within the hierarchy of chains
+func (progpow *Progpow) CalcOrder(header *types.Header) (*big.Int, int, error) {
```

Comment:
even POEM file I think, should be agnostic of the consensus engine

Reply:
POEM cannot calc intrinsic entropy (and by extension anything else) without adapting to the specific PoW being done. e.g. `Blake3pow` work is reflected in the block hash. `Progpow` work is reflected in a different pow hash. Thus POEM implementation on `Blake3pow` would look different than POEM implementation on `Progpow`.